### DISCUSSION TITLE
skip_dir question

### DISCUSSION THREAD

**Discussion by dtto** [abraunegg/onedrive#1840] [Q&A] [answered]
In my use case, there is a business shared folder that I need to sync called "All Staff" but a folder within the business share folder that I want to exclude. I have tried varies iterations of patterns in the config file for skip_dir but none so far have worked. So far, I've tried all of the following and none have worked so far. That is to say, the files still sync:

```
skip_dir = "./All Staff/Client Files/"
skip_dir = "./All Staff/Client Files"
skip_dir = "/All Staff/Client Files*"
skip_dir = "All Staff/Client Files*"
skip_dir = "Client Files*"
```

`onedrive --display-config` reflects the correct skip_dir listed in the config file, but nonetheless, I see (for all of the above):

`Adding 26 OneDrive items for processing from /All Staff/Client Files/`

etc.

What am I missing? I can put this into an issue if it's a bug, but I'm wondering if I'm just misunderstanding something basic about the way the skip_dir is supposed to work, or if there is something different needed since this is a business shared folder.

Thank you so much for your work on this software!

**Comment by abraunegg** [comment, marked as answer]
@dtto 
When the application is outputting `Adding 26 OneDrive items for processing from /All Staff/Client Files/` - it means it is checking OneDrive that particular path. It is adding the item path for validation.

Now - post this - is the application actually downloading data from that location or is it actually being skipped as per directed?

If they are being skipped, nothing will be downloaded.

If they are being downloaded - then this is a bug.

**Reply by dtto** [reply]
> Now - post this - is the application actually downloading data from that location or is it actually being skipped as per directed?

Honestly, I did not let it finish. I will check it now. Before I do, can you comment on which of the `skip_dir` formats I indicated in my original post would be ideal, or would they all work? It takes several hours to sync so it'd be great to know what the correct syntax is beforehand if possible. I can then test and, if necessary, report a bug if it is not working as intended.

Thank you again!

**Reply by abraunegg** [reply]
@dtto 
> Before I do, can you comment on which of the skip_dir formats I indicated in my original post would be ideal, or would they all work?

They all should work with the same effect

>  It takes several hours to sync so it'd be great to know what the correct syntax is beforehand if possible.

Always test by adding `--dry-run` to your command before doing an actual sync - this confirms your configuration and what would occur, without actually any actions being taken - ie - *much* quicker ......
